include/seqan3/search/search_result.hpp

```diff
@@ -0,0 +1,200 @@
+    friend class search_result_range;
+    //!\endcond
+
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    search_result() = default; //!< Defaulted.
+    search_result(search_result const &) = default; //!< Defaulted.
+    search_result(search_result &&) = default; //!< Defaulted.
+    search_result & operator=(search_result const &) = default; //!< Defaulted.
+    search_result & operator=(search_result &&) = default; //!< Defaulted.
+    ~search_result() = default; //!< Defaulted.
+    //!\}
+
+    /*!\name Accessors
+     * \brief Functions to access elements of the search result.
+     * \{
+     */
+    //!\brief Returns the id of the query which produced this search result.
+    constexpr query_id_type query_id() const noexcept
+    {
+        return query_id_;
+    }
+
+    /*!\brief Returns the index cursor pointing to the suffix array range where the query was found.
+     * \sa seqan3::fm_index_cursor
+     * \sa seqan3::bi_fm_index_cursor
+     */
+    constexpr cursor_type cursor() const noexcept(!(std::same_as<cursor_type, detail::empty_type>))
+    {
+        if constexpr (std::same_as<cursor_type, detail::empty_type>)
+        {
+            throw std::logic_error{"You tried to access the index cursor but it was not selected in the output "
+                                   "configuration of the search."};
+        }
+        return cursor_;
+    }
+
+    /*!\brief Returns the reference id where the query was found.
+     *
+     * \details
+     * The reference id is an arithmetic value that corresponds to the index of the reference text in the index.
+     * The order is determined on construction of the index.
+     */
+    constexpr reference_id_type reference_id() const noexcept(!(std::same_as<reference_id_type, detail::empty_type>))
+    {
+        if constexpr (std::same_as<reference_id_type, detail::empty_type>)
+        {
+            throw std::logic_error{"You tried to access the reference id but it was not selected in the output "
+                                   "configuration of the search."};
+        }
+        return reference_id_;
+    }
+
+    //!\brief Returns the reference begin positions where the query was found in the reference text (at `reference id`).
+    constexpr reference_begin_pos_type reference_begin_pos() const
+        noexcept(!(std::same_as<reference_begin_pos_type, detail::empty_type>))
+    {
+        if constexpr (std::same_as<reference_begin_pos_type, detail::empty_type>)
+        {
+            throw std::logic_error{"You tried to access the reference begin position but it was not selected in the "
+                                   "output configuration of the search."};
+        }
+        return reference_begin_pos_;
+    }
+    //!\}
+
+    /*!\name Comparison
+     * \{
+     */
+    //!\brief Returns whether `lhs` and `rhs` are the same.
+    friend bool operator==(search_result const & lhs, search_result const & rhs) noexcept
```

Comment:
Why do we need this?

Reply:
Because in the search test we sometimes compare a range of search results.